contracts/utils/TimestampedHashRegistry.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./interfaces/ITimestampedHashRegistry.sol";
+
+// TODO: would SelfMulticall be useful?
+contract TimestampedHashRegistry is Ownable, EIP712, ITimestampedHashRegistry {
+    using ECDSA for bytes32;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    mapping(bytes32 => EnumerableSet.AddressSet) private _hashTypeToSigners;
+
+    mapping(bytes32 => SignedHash) public hashTypeToSignedHash;
+
+    bytes32 private constant _SIGNED_HASH_TYPE_HASH =
+        keccak256(
+            "SignedHash(bytes32 hashType,bytes32 hash,uint256 timestamp)"
```

Comment:
If it's `hashType`, why is it not `hashTimestamp`? Since `hashTimestamp` sounds wrong, should it be `SignedHash(bytes32 type,bytes32 hash,uint256 timestamp)`?

Reply:
or `typeName` similar to `dapiName`?

Reply:
Reserved keyword collisions are avoided by a trailing `_` as per the style guide, but if it's actually a reserved keyword forget about it

Reply:
`typeName` is not correct because as I've mentioned, this is not necessarily a string or the hash of a string